### PR TITLE
Enable GitHub Actions concurrency feature to avoid queueing too many cross-version-tests runs

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Enable GitHub Actions concurrency feature to avoid queueing too many cross-version-tests runs.

## How is this patch tested?

Verify the concurrency feature works as expected.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/cross-version-tests.yml [hunk 1]
```diff
@@ -18,6 +18,10 @@ on:
     branches:
       - master
 
+concurrency:
+  group: cross-version-tests-${{ github.ref }}
+  cancel-in-progress: true
+
```
Comment (harupy) [inline .github/workflows/cross-version-tests.yml:24]: https://github.com/vitessio/vitess/pull/8546 describes well how the concurrency feature work:

> When a new commit is added to a pull request, GitHub Actions will add each workflow to groups. Groups are defined like this: branch_name + workflow_name, thus every workflow of the pull request's branch will be in their own unique group. GitHub Actions concurrency feature uses these groups to cancel workflow when new workflows (due to a new commit being pushed) are added to the queue. If for a given group there is a more recent workflow in the queue, the older workflows of the same group will be canceled by GitHub Actions, and only the most recent one will be executed.

**Comment by harupy** [comment]
I pushed three commits:

1. https://github.com/mlflow/mlflow/pull/4621/commits/abd068f7cd3ab014e8327245eacc8d2daf88f199
2. https://github.com/mlflow/mlflow/pull/4621/commits/cdbc3ed2cfce6a1aca4ef06eb7cc6475b7a3e86c
3. https://github.com/mlflow/mlflow/pull/4621/commits/4a6df64ff0939eaa9abb5eed86c54d5864b3f1b8

and confirmed:

The cross-version-tests run in the first commit was canceled by the second commit.
The cross-version-tests run in the second commit was canceled by the third commit.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy , awesome stuff!

**Comment by harupy** [comment]
@dbczumar Thanks for the reivew!

**Pull request merged**